django/db/models/functions.py

```diff
@@ -83,6 +83,28 @@ def _paired(self, expressions):
         return ConcatPair(expressions[0], self._paired(expressions[1:]))
 
 
+class Greatest(Func):
+    """
+    Chooses the maximum non-null expression and returns it.
+    """
+    function = 'GREATEST'
+
+    def __init__(self, *expressions, **extra):
+        if len(expressions) < 1:
+            raise ValueError('Greatest must take at least one expression')
+        super(Greatest, self).__init__(*expressions, **extra)
+
+    def as_sqlite(self, compiler, connection):
+        self.coalesce()
+        return super(Greatest, self).as_sql(compiler, connection, function='MAX')
+
+    def coalesce(self):
+        # any null results in null for the whole expression, wrap with coalesce
+        expressions = [
+            Coalesce(expression, Value('')) for expression in self.get_source_expressions()]
+        self.set_source_expressions(expressions)
```

Comment:
I agree it's awkward. It would be nice if we could automatically provide a sensible fallback to pass to `Coalesce`, but I think this is so context-dependent that we'd need to allow the user to override it anyway. I also think it will be difficult to get exact parity across databases if we want to handle `null` in a nice way.

Requiring the user to always provide a fallback value would be consistent, but this is awkward and may well not be what the user wants.